src/EnergyPlus/DaylightingManager.cc

```diff
@@ -7606,48 +7605,48 @@ void DayltgInterReflectedIllum(EnergyPlusData &state,
-                    DayltgInterReflectedIllumGroundHitPt.x = state.dataSurface->SurfaceWindow(IWin).WinCenter.x + HorDis * std::cos(Beta);
-                    DayltgInterReflectedIllumGroundHitPt.y = state.dataSurface->SurfaceWindow(IWin).WinCenter.y + HorDis * std::sin(Beta);
+                    groundHitPt.z = state.dataSurface->GroundLevelZ;
+                    groundHitPt.x = state.dataSurface->SurfaceWindow(IWin).WinCenter.x + HorDis * std::cos(Beta);
+                    groundHitPt.y = state.dataSurface->SurfaceWindow(IWin).WinCenter.y + HorDis * std::sin(Beta);
 
-                    SkyObstructionMult(IPH, ITH) = CalcObstrMultiplier(
-                        state, DayltgInterReflectedIllumGroundHitPt, AltAngStepsForSolReflCalc, DataSurfaces::AzimAngStepsForSolReflCalc);
+                    SkyObstructionMult[IPH][ITH] = CalcObstrMultiplier(
+                        state, groundHitPt, AltAngStepsForSolReflCalc, DataSurfaces::AzimAngStepsForSolReflCalc);
                 } // End of check if solar reflection calc is in effect
 
                 auto const &gilsk = state.dataDaylightingManager->GILSK(IHR);
                 for (int iSky = (int)SkyType::Clear; iSky < (int)SkyType::Num; ++iSky) {
                     // Below, luminance of ground in cd/m2 is illuminance on ground in lumens/m2
                     // times ground reflectance, divided by pi, times obstruction multiplier.
-                    ZSK.sky[iSky] = (gilsk.sky[iSky] * state.dataEnvrn->GndReflectanceForDayltg / Constant::Pi) * COSB * DA * ObTransM(IPH, ITH) *
-                                    SkyObstructionMult(IPH, ITH);
+                    ZSK.sky[iSky] = (gilsk.sky[iSky] * state.dataEnvrn->GndReflectanceForDayltg / Constant::Pi) * COSB * DA * ObTransM[IPH][ITH] *
+                                    SkyObstructionMult[IPH][ITH];
                 }
                 // Determine if sun illuminates the point that ray hits the ground. If the solar reflection
                 // calculation has been requested (CalcSolRefl = .TRUE.) shading by obstructions, including
                 // the building itself, is considered in determining whether sun hits the ground point.
                 // Otherwise this shading is ignored and the sun always hits the ground point.
                 SunObstructionMult = 1.0;
-                if (state.dataSurface->CalcSolRefl && ObTransM(IPH, ITH) > 1.e-6) {
+                if (state.dataSurface->CalcSolRefl && ObTransM[IPH][ITH] > 1.e-6) {
                     // Sun reaches ground point if vector from this point to the sun is unobstructed
                     hitObs = false;
                     for (int ObsSurfNum : state.dataSurface->AllShadowPossObstrSurfaceList) {
-                        PierceSurface(state, ObsSurfNum, DayltgInterReflectedIllumGroundHitPt, SUNCOS_IHR, DayltgInterReflectedIllumObsHitPt, hitObs);
+                        PierceSurface(state, ObsSurfNum, groundHitPt, SUNCOS_IHR, obsHitPt, hitObs);
```

Comment:
`integration._ExternalInterface-functionalmockupunit-to-schedule` is failing in `PierceSurface` because `groundHitPt` is NaNs. There may be a reason these were in `state`?

Reply:
Maybe. That particular state variable used to be called `DayltgInterReflectedIllumGroundHitPt` where `DayltgInterReflectedIllum` is the name of the function (the name of the function was prepended to the name of the local variable for disambiguation purposes because local variable names can be repeated across functions). That original state variable was only used in this function meaning that if this were really a state issue, `groundHitPt` was being passed from one called to this function to the next. Is that really the case here? It's more likely that it's a logic error that allows it to get this point without `groundHitPt` being initialized. 

I tried to make a non-FMU version of this file by removing the two OtherEquipment objects and that ran fine here, so I am not sure what the issue here or how to go about figuring it out. Maybe I can put that thing back in `state` and see.

Reply:
Should the condition on line 7628 also include `&& ISunPos == 1` to match the condition that sets `groundHitPt` on line 7599? That feels like a more likely culprit than moving this variable out of `state`.

Reply:
I'm not convinced that it's not "being passed from one called to this function to the next.'  `SkyObstructionMult` for example is set only when `ISunPos == 1` but used later. 

Looking back in the archives, [`SkyObstructionMult` was marked `SAVE`](https://github.com/NREL/EnergyPlusArchive/blob/92ffc69755ab1ed2315a7a49a9b04fb000aecb25/SourceCode/DaylightingManager.f90#L7531) so it should remain in `state`. [`ObsHitPt`](https://github.com/NREL/EnergyPlusArchive/blob/92ffc69755ab1ed2315a7a49a9b04fb000aecb25/SourceCode/DaylightingManager.f90#L7538) was not `SAVE`d, so I'm convinced now.

There are no diffs or failures now, so this looks good. @Myoldmopar Tag - final review/merge is to you.

Reply:
I am not convinced the historic variables that were marked SAVE are still working the same way. I have seen at least 1 that I thought should no longer work as intended. I had a conversation with @Myoldmopar on a CppCheck branch on this subject so I could probably find the variable in question.